source/common/event/scaled_range_timer.h

```diff
@@ -0,0 +1,69 @@
+#include <chrono>
+
+#include "envoy/event/timer.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace Event {
+
+class ScaledRangeTimerManager {
+public:
+  ScaledRangeTimerManager(
+      Dispatcher& dispatcher, float scale_factor,
+      std::chrono::milliseconds minimum_duration = std::chrono::milliseconds(100));
+
+  RangeTimerPtr createTimer(TimerCb callback);
+
+  void setScaleFactor(float scale_factor);
```

Comment:
Should you take a OverloadActionState as input instead?

The thing to note is that I think that what you're calling scale_factor here is: 1.0 - OverloadActionState

Reply:
I plan to use this in the overload manager but there's no reason to tie them together. #12352 also hasn't landed yet.